.github/workflows/ci.yml

```diff
@@ -635,8 +635,11 @@ jobs:
             bash -euo pipefail -c \
             "./build_tools/cmake/build_riscv.sh && tests/riscv32/smoke.sh"
 
+  # TODO(#10391): Update the benchmark dependency with the new benchmark
+  # framework. The step only requires the compiled benchmark modules, not the
+  # not the result from the benchmark.
   linux_riscv:
-    needs: [setup, build_all, build_tf_integrations]
+    needs: [setup, build_all, benchmarks]
```

Comment:
We should be testing program correctness on all supported platforms, not just riscv. Perhaps a `matrix` of platforms, for a new `test_model_suites` job? I want us to avoid needing to duplicate all this "download, extract, set environment variables" boilerplate for each platform.

```
build_all, build_tf_integrations -> build_model_suites (for running benchmarks and correctness tests)

build_tools_[platform], build_model_suites -> test_model_suites_[platform]
build_tools_[platform], build_model_suites -> benchmark_model_suites_[platform]
```

Reply:
Some of the boilerplate code will be needed in the cross-compile build_x.sh, but I agree with the approach. I can rework https://github.com/iree-org/iree/pull/10652 to address that.

Reply:
It might take a few steps to get there, and testing on riscv like this will provide more immediate value, I just want to make sure we have a clear picture of the architecture (for the multi-stage CI pipelines) we're aiming for.

Reply:
Understood. I think the following steps would be necessary

- [ ] Add `benchmark-suite-test` build preset for all x-compile (and host) platforms
- [ ] Split up `runtime` and regular `tests/` ctests with benchmark-suite-module test in `test_riscv.sh`, and create another matrix for the regular build/test
- [ ] Add android-arm64 into both jobs. However, we need to figure out how/if we can run the ctest to check the correctness, or just build the testing script for the time being.